internal/credential/state.go

```diff
@@ -20,9 +20,23 @@ import (
 type CredentialType int
 
 const (
-	Unknown CredentialType = iota
-	Static
-	Dynamic
+	// StaticAWS denotes an Access Key Id that begins with "AKIA". These are
+	// long-term access keys, provided by AWS, for an IAM user or an AWS account
+	// root user.
+	// https://docs.aws.amazon.com/IAM/latest/UserGuide/security-creds.html#sec-access-keys-and-secret-access-keys
+	StaticAWS CredentialType = iota
+
+	// DynamicAWS denotes the presence of a RoleARN, or an Access Key Id that
+	// begins with "ASIA". The latter are temporary credentials access keys that
+	// are created using AWS STS operations.
+	DynamicAWS
+
+	// StaticOther denotes the presence of an Access Key Id that does not follow
+	// the AKIA/ASIA convention outlined by AWS.
+	StaticOther
+
+	// Unknown is a catch-all for everything else.
+	Unknown
```

Comment:
What would the behavior of unknown be?

Reply:
For the most part, it's used to carve out a special condition (eg: `ToMap`) or an error, since we don't really know what it is.